plugins/ccgx/fu-ccgx-hid-device.c

```diff
@@ -44,13 +45,47 @@ fu_ccgx_hid_device_detach (FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_ccgx_hid_device_setup_cb (gpointer user_data)
+{
+	FuCcgxHidDevice *self = FU_CCGX_HID_DEVICE (user_data);
+	g_autoptr(GError) error = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* do this in idle so that the original HID device gets added */
+	locker = fu_device_locker_new (self, &error);
+	if (locker == NULL) {
+		g_warning ("failed to open HID device: %s",
+			   error->message);
+	} else {
+		if (!fu_ccgx_hid_device_detach (FU_DEVICE (self), &error)) {
```

Comment:
What actually happens?  Does the FuDevice disappear?  I worry this is a timing hack that might not behave the same reliably.

Reply:
Yes, it unplugs and replugs with a different VID PID.

Reply:
actually, maybe a cleaner thing to do might be to detach and then return FALSE...

Reply:
actually 2, we need to add the device sometimes as in future hardware the HID mode is updatable, so we can't return failure from setup.

Reply:
Well you could also just have a dummy setup method and do the detach in probe perhaps?

Reply:
I've switched it to returning FALSE from setup. Setup already has the device open so it's the natural time to do it. If/when we have a device that can be flashing in HID mode we can just do a different thing in setup.